src/ethereum_test/types.py

```diff
@@ -257,24 +342,58 @@ def __post_init__(self) -> None:
             and self.max_fee_per_gas is None
             and self.max_priority_fee_per_gas is None
         ):
-            self.gas_price = 0
+            self.gas_price = 10
 
         if self.signature is not None and self.secret_key is not None:
             raise Exception("can't define both 'signature' and 'private_key'")
 
         if self.signature is None and self.secret_key is None:
             self.secret_key = TestPrivateKey
 
+        if self.ty is None:
+            # Try to deduce transaction type from included fields
+            if self.max_fee_per_gas is not None:
+                self.ty = 2
+            elif self.access_list is not None:
+                self.ty = 1
+            else:
+                self.ty = 0
+
 
-@dataclass
+@dataclass(kw_only=True)
 class Header:
     """
-    Ethereum header object.
+    Header type used to describe block header properties in test specs.
+    """
+
+    coinbase: Optional[str] = None
+    parent_hash: Optional[str] = None
+    ommers_hash: Optional[str] = None
+    state_root: Optional[str] = None
+    transactions_root: Optional[str] = None
+    receipt_root: Optional[str] = None
+    bloom: Optional[str] = None
+    difficulty: Optional[int] = None
+    number: Optional[int] = None
+    gas_limit: Optional[int] = None
+    gas_used: Optional[int] = None
+    timestamp: Optional[int] = None
+    extra_data: Optional[str] = None
+    mix_digest: Optional[str] = None
+    nonce: Optional[str] = None
+    base_fee: Optional[int] = None
+    hash: Optional[str] = None
+
+
+@dataclass(kw_only=True)
+class FixtureHeader:
+    """
+    Representation of an Ethereum header within a test Fixture.
     """
 
+    coinbase: str
     parent_hash: str
     ommers_hash: str
-    coinbase: str
```

Comment:
What is the rationale for moving this? It's kind of nice that it is currently in the RLP list order.

Reply:
This was a mistake, I moved temporarily for mental order while I was refactoring but forgot to put it back. Changing it bacl now.